### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by theseal**
* Show the last X-forwarded-for header in logs (instead of first)
* Replace own logging function with hlog library (thanks @eest)
* * This will make the logs more of a truth then a guess
* * Start logging as json for easier parsing
* A few small general tweaks and improvements

### CODE REVIEW

**Comment by theseal** [comment]
Tested in QA for the last 12 hours and seems to run just fine.

File: build/publisher.go [hunk 1]
```diff
@@ -22,32 +24,27 @@ type myMux struct {
...
-		requestor = splitAddr[0]
-	} else {
-		requestor = "UNKNOWN"
+	xffs := req.Header["X-Forwarded-For"]
+	if len(xffs) > 0 {
+		// From HAProxy's documentation:
+		// Since this header is always appended at the end of the existing header list, the server must be configured to always use the last occurrence of this header only.
+		xff := xffs[len(xffs)-1]
+		logger.UpdateContext(func(c zerolog.Context) zerolog.Context {
+			return c.Str("x_forwarded_for", xff)
```
Comment (mikaelfrykholm) [inline build/publisher.go:39]: Nice, an even more nice sollution would be to use the scheme mozilla suggests here:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For
ie. join all x-forwarded-for with a comma.

**Review by mikaelfrykholm — CHANGES_REQUESTED** [review]
Great work, please take a look at my comment if that is easily doable.

File: build/publisher.go (hunk 1 above)
Comment (theseal) [reply build/publisher.go:39]: It's doable. But do we need it?

I don't think we should cover it in this PR.

**Review by btmattsson — APPROVED** [review]

**Pull request merged**